tests/unit/projection-test.js

```diff
@@ -2224,6 +2235,82 @@ module('unit/projection', function(hooks) {
       assert.equal(get(projectedPreview, 'title'), BOOK_TITLE);
     });
 
+    skip(
+      'Unloading a record removes it from record arrays, which have reference to it',
+      function(assert) {
+        // we need additional records to be able to resolve the references
+        run(() => {
+          this.store.push({
+            data: {
+              id: OTHER_BOOK_ID,
+              type: BOOK_CLASS_PATH,
+              attributes: {},
+            },
+          });
+        });
+
+        run(() => {
+          this.store.push({
+            data: {
+              id: OTHER_BOOK_ID,
+              type: BOOK_PREVIEW_PROJECTION_CLASS_PATH,
+            },
+          });
+        });
+
+        let { baseModel, projectedPreview } = this.records;
+
+        // load the record arrays
+        let booksInSeriesBase = get(baseModel, 'otherBooksInSeries');
+        let booksInSeriesProjectedPreview = get(
+          projectedPreview,
+          'otherBooksInSeries'
+        );
+        let otherProjectedPreview = get(
+          booksInSeriesProjectedPreview,
+          'firstObject'
+        );
+
+        // precondition
+        assert.equal(
+          get(booksInSeriesBase, 'length'),
+          1,
+          'Expected otherBooksInSeries length to be one for base'
+        );
+        assert.equal(
+          get(booksInSeriesProjectedPreview, 'length'),
+          1,
+          'Expected otherBooksInSeries length to be one for projected preview'
+        );
+
+        // unload a projection referenced in a record array
+        run(() => {
+          otherProjectedPreview.unloadRecord();
+        });
+
+        assert.equal(
+          get(booksInSeriesBase, 'length'),
+          1,
+          'Expected otherBooksInSeries length to be unchanged for base'
+        );
+        assert.equal(
+          get(booksInSeriesProjectedPreview, 'length'),
+          1,
+          'Expected otherBooksInSeries length to be unchanged for projected preview'
+        );
+        assert.equal(
+          booksInSeriesProjectedPreview.getObjectAt(0),
+          null,
```

Comment:
why do we expect `null` instead of `undefined`? ember-data bug?

Reply:
The `assert.equal` is not strict, therefore `null` or `undefined` are both valid results for the purpose of the test. Ideally, the value should be synchronized with the result from `peekRecord` in case of missing record.

Reply:
No, we should assert here.  The analog is ember-data sync relationships, in which case attempts to retrieve non-loaded records results in an assertion failure.

Reply:
@hjdivad The current behavior is required for the following use case:
- We fetch an array with specific projection
- We refetch the array with a different projection
- The refetch may have added new elements, which however were fetched with different projection, therefore the original references will now attempt to reference a record, which doesn't have the required projection.

The current agreement for this case is to handle it gracefully (either by the app or ember-m3) by filtering out records from the array, which doesn't have the required projections.